tests/e2e/test_pgsql_node_reboot.py

```diff
@@ -0,0 +1,139 @@
+"""
+Module to perform PGSQL workload
+"""
+import logging
+import pytest
+from ocs_ci.ocs.resources.ocs import OCS
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.utility import templating, utils
+from ocs_ci.utility.utils import run_cmd, TimeoutSampler
+from ocs_ci.ocs.utils import get_pod_name_by_pattern
+from ocs_ci.ocs.ripsaw import RipSaw
+from ocs_ci.ocs import constants
+from ocs_ci.ocs import node
+from tests.sanity_helpers import Sanity
+from ocs_ci.framework.testlib import (
+    E2ETest, google_api_required, workloads, ignore_leftovers
+)
+from ocs_ci.ocs.exceptions import UnexpectedBehaviour
+from ocs_ci.utility.spreadsheet.spreadsheet_api import GoogleSpreadSheetAPI
+
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope='function')
+def ripsaw(request):
+
+    # Create RipSaw Operator
+    ripsaw = RipSaw()
+
+    def teardown():
+        ripsaw.cleanup()
+    request.addfinalizer(teardown)
+    return ripsaw
+
+
+@ignore_leftovers
+@workloads
+@google_api_required
+@pytest.mark.polarion_id("OCS-806")
+class TestPgSQLNodeReboot(E2ETest):
+    """
+    Test running PGSQL and with Ceph pods respin
+    """
+    @pytest.fixture()
+    def pgsql_setup(self, ripsaw):
+        """
+        PGSQL test setup
+        """
+        # Deployment ripsaw and postgres database
+        log.info("Deploying postgres database")
+        ripsaw.apply_crd(
+            'resources/crds/'
+            'ripsaw_v1alpha1_ripsaw_crd.yaml'
+        )
+        ripsaw.setup_postgresql()
+
+        # Initialize Sanity instance
+        self.sanity_helpers = Sanity()
+
+    @pytest.mark.usefixtures(pgsql_setup.__name__)
+    def test_run_pgsql_node_reboot(self, transactions=900, node_type='master'):
+        """
+        Test pgsql workload
+        """
+        # Create pgbench benchmark
+        log.info("Create resource file for pgbench workload")
+        pg_trans = transactions
+        timeout = pg_trans * 3
+        pg_data = templating.load_yaml(constants.PGSQL_BENCHMARK_YAML)
+        pg_data['spec']['workload']['args']['transactions'] = pg_trans
+        pg_obj = OCS(**pg_data)
+        pg_obj.create()
+
+        # Wait for pgbench pod to be created
+        for pgbench_pod in TimeoutSampler(
+            pg_trans, 3, get_pod_name_by_pattern,
+            'pgbench', 'my-ripsaw'
+        ):
+            try:
+                if pgbench_pod[0] is not None:
+                    pgbench_client_pod = pgbench_pod[0]
+                    break
+            except IndexError:
+                log.info("Bench pod not ready yet")
+
+        # Node reboot with specific node type
+        typed_nodes = node.get_typed_nodes(node_type=node_type, num_of_nodes=1)
+        typed_node_name = typed_nodes[0].name
+
+        # Node maintenance - to gracefully terminate all pods on the node
+        node.drain_nodes([typed_node_name])
+
```

Comment:
I think it will be good idea to put here sleep for few (1-3) second.

Reply:
Drain_nodes should have wait function with it, if not then its better to impvore that function there

Reply:
Support function already has wait time.  I don't think we need to add sleep time here.